ci/ray_ci/macos/macos_ci_build.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -ex
+
+export CI="true"
+export PYTHON="3.8"
+export RAY_USE_RANDOM_PORTS="1"
+export RAY_DEFAULT_BUILD="1"
+export LC_ALL="en_US.UTF-8"
+export LANG="en_US.UTF-8"
+export BUILD="1"
+export DL="1"
+
+
+build_x86_64() {
+  # Cleanup environments
+  rm -rf /tmp/bazel_event_logs
+  cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
+  (which bazel && bazel clean) || true
+  # TODO(simon): make sure to change both PR and wheel builds
+  # Special setup for jar builds (will be installed to the machine instead)
+  # - brew remove --force java & brew uninstall --force java & rm -rf /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask
+  # - brew install --cask adoptopenjdk/openjdk/adoptopenjdk8
+  diskutil list external physical
+  export JAVA_HOME=/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
+  java -version
+  # Build wheels
+  export UPLOAD_WHEELS_AS_ARTIFACTS=1
+  export MAC_WHEELS=1
+  export MAC_JARS=1
+  export RAY_INSTALL_JAVA=1
+  export RAY_ENABLE_WINDOWS_OR_OSX_CLUSTER=1
+  . ./ci/ci.sh init && source ~/.zshenv
+  source ~/.zshrc
+  ./ci/ci.sh build
+  # Test wheels
+  ./ci/ci.sh test_wheels
+  # Build jars
+  bash ./java/build-jar-multiplatform.sh darwin
+  # Upload the wheels and jars
+  # We don't want to push on PRs, in fact, the copy_files will fail because unauthenticated.
+  if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then exit 0; fi
```

Comment:
check if it is on postmerge pipeline here? I do not want it to depend on if it is pull request triggered.